packages/toolkit/src/components/card-model/CardModel.tsx

```diff
@@ -0,0 +1,111 @@
+"use client";
+
+import Image from "next/image";
+import { Menu } from "./Menu";
+import { Tags } from "./Tags";
+import { Stats } from "./Stats";
+import {
+  ListUserModelsResponse,
+  Model,
+  Nullable,
+  sendAmplitudeData,
+  useAmplitudeCtx,
+  useDeleteModel,
+} from "../../lib";
+import React from "react";
+import {
+  useToast,
+  getModelHardwareToolkit,
+  getModelRegionToolkit,
+} from "@instill-ai/design-system";
+import axios from "axios";
+import { useParams } from "next/navigation";
+import {
+  InfiniteData,
+  QueryObserverResult,
+  RefetchOptions,
+} from "@tanstack/react-query";
+
+export type CardModelProps = {
+  model: Model;
+  accessToken: Nullable<string>;
+  onDelete: (
+    options?: RefetchOptions | undefined
+  ) => Promise<
+    QueryObserverResult<InfiniteData<ListUserModelsResponse, unknown>, Error>
+  >;
```

Comment:
@orangecoloured I think this callback return is a bit confusing, I noticed that you try to use `query.refetch`

We can make it more general, just like (options) => void

and you use it as `() => query.refetch`

But if you want this prop to be only used for refetch then we should rename it to be more specific. I am ok with either way 🦤

Reply:
I simplified it to `() => void`. If we need to pass any options we can always update the type.